.github/workflows/HardhatTests.js.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, and run hardhat tests on Arrow contracts
+
+name: Hardhat Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # run this workflow every Monday at 1PM UTC
+    - cron: "* 13 * * 1"
```

Comment:
```suggestion
```
I don't think there's any need to run this weekly, as well as on PRs and pushes to branches, because the code doesn't change weekly.